LearningMissionSimulation/PlaygroundVahe.cs

```diff
@@ -156,17 +247,177 @@ List<Module> GetModuleList()
 
         public void CreateClassrooms(int classroomCount)
         {
-            throw new NotImplementedException();
+            string action = "Create classrooms";
+            ReportHeader(action);
+
+            if (ModuleIdList.Count != 0)
+            {
+                int i = 0;
+                while (i < classroomCount)
+                {
+                    Guid moduleId = ModuleIdList[AttributeGenerator.random.Next(0, ModuleIdList.Count)];
+                    Module module;
+                    ModuleDictionary.TryGetValue(moduleId, out module);
+                    Classroom classroom = ObjectGenerator.GenerateClassroom(module);
+                    ClassroomDictionary.Add(classroom.Id, classroom);
+                    ReportItem(classroom.ToString(), action, i++);
+                }
+            }
+            else
+            {
+                ReportError(ModuleIdList.ToString(), action);
+            }
+            ReportSummary(action, classroomCount, "classroom");
+            ReportFooter(action);
+
         }
-        
+
         public void AssignInstructorsToClassrooms()
         {
-            throw new NotImplementedException();
+            string action = "Assign instructor to classroom";
+            ReportHeader(action);
+
+            if (ActiveInstructorList.Count > 0 && ClassroomDictionary.Count > 0)
+            {
+                foreach (var classroom in ClassroomDictionary.Values)
+                {
+                    Guid moduleId = classroom.Module.Id;
+                    List<Instructor> instructorList;
+                    if (ModuleInstructorListDictionary.ContainsKey(moduleId))
+                    {
+                        ModuleInstructorListDictionary.TryGetValue(moduleId, out instructorList);
+                        classroom.Head = instructorList[AttributeGenerator.random.Next(0, instructorList.Count)];
+                    }
```

Comment:
You should ad an **else ** block here and report an error if the moduleId is not found and you can't assign an instructor to the class.